clusterctl/examples/google/generate-yaml.sh

```diff
@@ -21,7 +21,9 @@ set -o pipefail
 GCLOUD_PROJECT=$(gcloud config get-value project)
 ZONE=$(gcloud config get-value compute/zone)
 ZONE="${ZONE:-us-central1-f}"
-CLUSTER_NAME=test1
+# Generate a somewhat unique cluster name, UUID is not an option as service-accounts are limited to 30 characters in length
+# and have a 19 character prefix (i.e. 'machine-controller-')
+CLUSTER_NAME=$(head -c11 < <(LC_ALL=C tr -dc 'a-zA-Z0-9' < /dev/urandom) | tr '[:upper:]' '[:lower:]')
```

Comment:
I think it'd be better to concatenate a human friendly name and a random string segment. Especially if you are trying to do things like manually clean up service accounts in the cloud console, it's much easier to think about looking for "test1" and just kill everything with that prefix.

Reply:
Yeah I debated that and went with all random due to the limited number of characters that we can use. 

I'll add back in the friendly name with a comment about the restrictions on length.

Reply:
Note that we could possibly follow up with another PR where we make some of the names shorter, for example, the service account for machine controller currently named:

`machine-controller-${CLUSTER_NAME}`

Could be coming something like,

`m-control-${CLUSTER_NAME}`

I'd prefer to follow up with that in another PR as the limitation on cluster name length is not a new issue introduced by this PR (although I guess it is made a bit worse by the random string).